buildstockbatch/gcp/gcp.py

```diff
@@ -456,9 +456,8 @@ def show_jobs(self):
         Show the existing GCP Batch and Cloud Run jobs that match the provided project, if they exist.
         """
         # GCP Batch job that runs the simulations
-        client = batch_v1.BatchServiceClient()
-        try:
-            job = client.get_job(batch_v1.GetJobRequest(name=self.gcp_batch_job_name))
+        job = self.get_existing_batch_job()
+        if job:
```

Comment:
You can combine the assignment and `if`, which indicates/reflects that `job` is only used within the `if`:
```suggestion
        if job := self.get_existing_batch_job():
```

Reply:
Done.

I still think of this as a brand new Python feature that might not be compatible with everywhere this code needs to run... but I just looked it up and it's been around for years now. 🤦‍♀️

Reply:
It's nice that with venv & Docker we know what version of Python we'll be using.

Something that was only officially available 2 months ago (with the release of 3.12) that we can't use yet is the [override decorator](https://docs.python.org/3/whatsnew/3.12.html#pep-698-override-decorator-for-static-typing)! :disappointed: